lib/rspec/mocks/proxy.rb

```diff
@@ -434,6 +440,11 @@ def warn(method_name)
         source = CallerFilter.first_non_rspec_line
         Kernel.warn("An expectation of :#{method_name} was set on nil. Called from #{source}. Use allow_message_expectations_on_nil to disable warnings.")
       end
+
+      def raise_error(method_name)
+        source = CallerFilter.first_non_rspec_line
+        raise("An expectation of :#{method_name} was set on nil. Called from #{source}. To allow expectations on nil, set disallow_expectations_on_nil to false.")
```

Comment:
Please use `ErrorGenerator`.  It is intended to be the central place for errors to be generated (as the name implies) although that hasn't always been followed.  Having it delegate to `RSpec::Support.notify_failure` will be good to make this work with `aggregate_failures` and be treated as a normal rspec expectation failure.